kythe/java/com/google/devtools/kythe/analyzers/java/JavaEntrySets.java

```diff
@@ -124,6 +125,75 @@ public EntrySet getNode(Symbol sym, String signature) {
     return node;
   }
 
+  private static NodeBuilder setFlagProperties(Symbol sym, NodeBuilder builder) {
+    long flags = sym.flags();
+    switch(sym.getKind()) {
+      case ENUM:
+      case CLASS:
+      case ANNOTATION_TYPE:
+      case INTERFACE:
+        builder = builder.setProperty("access", accessFlag(flags));
+        builder = setPropertyIfFlag(builder, flags & Flags.STATIC, "static");
+        builder = setPropertyIfFlag(builder, flags & Flags.FINAL, "final");
+        builder = setPropertyIfFlag(builder, flags & Flags.ABSTRACT, "abstract");
+        break;
+      case METHOD:
+      case CONSTRUCTOR:
+        builder = builder.setProperty("access", accessFlag(flags));
+        builder = setPropertyIfFlag(builder, flags & Flags.STATIC, "static");
+        builder = setPropertyIfFlag(builder, flags & Flags.FINAL, "final");
+        builder = setPropertyIfFlag(builder, flags & Flags.SYNCHRONIZED, "synchronized");
```

Comment:
Many of these are java-specific concepts. Will they expect them to be generalized so they can be applied to all languages? Or should they be prefixed with "java/" (e.g., "java/synchronized") to emphasize that these are language specific properties? Do they have any facility for language-specific stuff?